File: design-pattern/src/main/java/com/orcl/design/composite/File.java
package com.orcl.design.composite;

/**
 * ファイルを表現するクラス
 */

public class File extends Entry{
    private int size;
    private String name;
    public File(String name,int size){
        this.name = name;
        this.size = size;
    }
    @Override
    public String getName(){
        return name;
    }
    @Override
    public int getSize(){
        return size;
    }
    @Override
    protected void printList(String prefix) {
        System.out.println(prefix + "/" + this.toString());
    }
}

File: cpp/gof-cpp/visitor/File.cpp
#include "Entry.cpp"
#include "Visitor.cpp"
#include <string>
class File : public Entry{
    private :
        std::string name;
        int size;
    public :
        File(std::string n, int s){
            this->name = n;
            this->size = s;
        };
    public :
        std::string getName() override{
            return name;
        };
        int getSize() override{
            return size;
        };
    
    public:
        virtual void accept(Visitor& v) override
        {
            v.visit(*this);
        };
};
